### PR TITLE
check cluster version on store tombstone

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?
Check and update cluster version on a store becomes tombstone.

Tests

- Unit test

Code changes

Side effects

Related changes

### Release note

* check cluster version on stores become tombstone

### CODE REVIEW

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -1042,6 +1042,7 @@ func (c *RaftCluster) BuryStore(storeID uint64, force bool) error {
 
 	// Bury a tombstone store should be OK, nothing to do.
 	if store.IsTombstone() {
+		c.onStoreVersionChangeLocked()
```
Comment (rleungx) [inline server/cluster/cluster.go:1045]: Do we need it?

**Review by rleungx — APPROVED** [review]
LGTM

**Review by HunDunDM — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]
LGTM

File: tests/server/cluster/cluster_test.go [hunk 2]
```diff
@@ -168,6 +168,45 @@ func (s *clusterTestSuite) TestGetPutConfig(c *C) {
 	c.Assert(meta.GetMaxPeerCount(), Equals, uint32(5))
 }
 
+func (s *clusterTestSuite) TestDeleteStoreUpdatesClusterVersion(c *C) {
+	tc, err := tests.NewTestCluster(s.ctx, 1)
+	defer tc.Destroy()
+	c.Assert(err, IsNil)
```
Comment (HunDunDM) [inline tests/server/cluster/cluster_test.go:174]: It is better to swap two lines.

File: tests/server/cluster/cluster_test.go [hunk 3]
```diff
@@ -168,6 +168,45 @@ func (s *clusterTestSuite) TestGetPutConfig(c *C) {
...
+	tc.WaitLeader()
+	leaderServer := tc.GetServer(tc.GetLeader())
+	grpcPDClient := testutil.MustNewGrpcClient(c, leaderServer.GetAddr())
+	clusterID := leaderServer.GetClusterID()
+	bootstrapCluster(c, clusterID, grpcPDClient, "127.0.0.1:1", "4.0.9")
+	rc := leaderServer.GetRaftCluster()
+	c.Assert(rc, NotNil)
+
+	// Put 2 new stores.
+	for _, id := range []uint64{2, 3} {
```
Comment (HunDunDM) [inline tests/server/cluster/cluster_test.go:188]:
```suggestion
	for _, id := range []uint64{200, 300} {
```

File: tests/server/cluster/cluster_test.go [hunk 4]
```diff
@@ -168,6 +168,45 @@ func (s *clusterTestSuite) TestGetPutConfig(c *C) {
...
+		addr := fmt.Sprintf("127.0.0.1:%d", id)
+		store := newMetaStore(id, addr, "4.0.9", metapb.StoreState_Up, fmt.Sprintf("test/store%d", id))
+		_, err = putStore(c, grpcPDClient, clusterID, store)
+		c.Assert(err, IsNil)
+	}
+	c.Assert(rc.GetClusterVersion(), Equals, "4.0.9")
+
+	// Upgrade 2 stores to 5.0.0.
+	for _, id := range []uint64{2, 3} {
```
Comment (HunDunDM) [inline tests/server/cluster/cluster_test.go:197]:
```suggestion
	for _, id := range []uint64{200, 300} {
```

**Comment by HunDunDM** [comment]
Is it a more reasonable choice to write a unit test directly under `pd/server/cluster/cluster_test.go`?

**Comment by HunDunDM** [comment]
I updated the path of unit test, @nolouch @rleungx @Yisaer PTAL

**Comment by hicqu** [comment]
LGTM for changes about the new test case. Thank you @HunDunDM .

**Pull request merged**